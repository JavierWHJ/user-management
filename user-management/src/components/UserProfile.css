.avatar-circle {
    width: 50px;
    height: 50px;
    background-color: #3E83F8;
    text-align: center;
    border-radius: 50%;
    -webkit-border-radius: 50%;
    -moz-border-radius: 50%;
}

.initials {
    position: relative;
    top: 20%; /* 25% of parent */
    font-size: 100%; /* 50% of parent */
    line-height: 100%; /* 50% of parent */
    color: #fff;
    font-family: "Courier New", monospace;
    font-weight: bold;
}