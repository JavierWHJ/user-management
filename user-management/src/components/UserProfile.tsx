import React from 'react';
import './UserProfile.css';

interface Props {
    firstName: string,
    lastName: string
}


const UserProfile = (props: Props) => {
    return (
        <div className="avatar-circle">
            <span className="initials">{(props.firstName[0] + props.lastName[0]).toUpperCase()}</span>
        </div>
    );
}

export default UserProfile;
