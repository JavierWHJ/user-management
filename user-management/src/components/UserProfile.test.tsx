import React from "react";
import UserProfile from "./UserProfile";

import { render } from "@testing-library/react";

it("matches snapshot", () => {
  const { asFragment } = render(<UserProfile firstName={"John"} lastName={"Doe"}/>);

  expect(asFragment()).toMatchSnapshot();
});