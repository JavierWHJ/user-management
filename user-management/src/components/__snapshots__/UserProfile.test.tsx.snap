// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="avatar-circle"
  >
    <span
      class="initials"
    >
      JD
    </span>
  </div>
</DocumentFragment>
`;
